library/ssl_srv.c

```diff
@@ -2969,11 +2969,9 @@ static int ssl_write_certificate_request( mbedtls_ssl_context *ssl )
 
         while( crt != NULL && crt->version != 0 )
         {
-            dn_size = crt->subject_raw.len;
+            dn_size = (uint16_t)crt->subject_raw.len;
 
-            if( end < p ||
-                (size_t)( end - p ) < dn_size ||
-                (size_t)( end - p ) < 2 + dn_size )
+            if( end < p || (size_t)( end - p ) < 2 + (size_t)dn_size )
```

Comment:
Style: we usually leave a space in `(size_t) dn_size` (we only omit it before a parenthesis).